pkg/frontend/frontend.go

```diff
@@ -41,6 +42,8 @@ func (err statusCodeError) Error() string {
 	return fmt.Sprintf("%d", err)
 }
 
+type hiveActionsFactory func(*logrus.Entry, env.Interface) (hive.ClusterManager, error)
```

Comment:
do we really need a factory here? these are not great , the others needed the factory because they were created with the openshift cluster as a parameter (and it's was a bad design), here I don't think we need it and it'd be great to not add more of those

Reply:
I did this because I think eventually this will be updated to include the OpenshiftCluster so we can know which shard to access. Otherwise we could just create the hiveclientset and add it as a part of the frontend without a factory.

Reply:
still, I don't think the factory is adding any value here, because 
1- the env is never changing so reconstructing it every time isn't really useful
2- the log is changing but we are only using it once so we can just add it as a parameter on the one function
3- we can add the openshiftcluster as an argument for the functions that need it

overall removing it is just going to simplify the code imho

Reply:
I agree with those points, I will directly inject the `hive.ClusterManager` to the frontend and get it pushed up.